testing/regress/hpcc/regression/cleanup.py

```diff
@@ -0,0 +1,94 @@
+import logging
+import os
+import re
+import requests
+import time
+from datetime import datetime
+
+# Configure logger
+logger = logging.getLogger('cleanup') 
+
+# Custom logger formatter for serial numbering of log records
+class SerialNumberFormatter(logging.Formatter):
+    def __init__(self):
+        super().__init__()
+        self.serialNumber = 0
+
+    def format(self, record):
+        self.serialNumber += 1
+        record.msg = str(self.serialNumber) + ". " + record.msg
+        return super().format(record)
+
+# Builds and configures the cleanup logger
+def buildCleanupLogger(logDir, cleanupLogger):
+    cleanupLogger.setLevel(logging.INFO)
+    curTime = time.strftime('%y-%m-%d-%H-%M-%S')
+    logName = "cleanup" + "." + curTime + ".log"
+    logPath = os.path.join(logDir, logName)
+    cleanupHandler = logging.FileHandler(logPath)
+    cleanupHandler.setFormatter(SerialNumberFormatter())
+    cleanupLogger.addHandler(cleanupHandler)
+    return cleanupLogger
+
+# Extracts regress log files corresponding to current run of RTE
+def getRegressLogs(mode, logDirPath, startTime):
+    logFileNamePattern = re.compile(r'^(thor|roxie|hthor|roxie-workunit)\.\d{2}-\d{2}-\d{2}-\d{2}-\d{2}-\d{2}\.log(?!.*exclusion)')
+    with os.scandir(logDirPath) as entries:
+        for entry in entries:
```

Comment:
If I remember well, we already discussed this. If I have hundreds or thousands logfiles this loop goes trough all of them, however you need to process only a couple of logs created recently and you have a start timestamp as well.

Reply:
We found an approach where we only look through the files created during the current run of query or run sub-command. It uses the Python watchdog library which needs to be pip installed to be used. May we please know if this is 
the approach?